experiments/functional/cstor-volume-migration/test.yml

```diff
@@ -0,0 +1,433 @@
+
+       - name: Obtain the new replica id value from cstorvolume
+         shell: >
+           kubectl get cstorvolume {{ cv_name.stdout }} -n {{ operator_ns }}
+           -o custom-columns=:.status.replicaDetails.knownReplicas.{{ replica_id.stdout }} --no-headers
+         args:
+           executable: /bin/bash
+         register: new_rid_value
+         until: "new_rid_value.stdout != rid_value.stdout"
+         retries: 30
+         delay: 10
+
+       - name: Obtain the ReplicationFactor after cvr migration and compare with replication factor obtained before migrate cvr
+         shell: >
+           kubectl get cstorvolume -n {{ operator_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ cv_name.stdout }}")].spec.replicationFactor}{"\n"}'
+         args:
+           executable: /bin/bash
+         register: after_rf
+         failed_when: "after_rf.stdout != rf.stdout"
+
+       - name: Obtain the consistencyFactor after cvr migration and compare with consistency factor obtained before migrate cvr
+         shell: >
+           kubectl get cstorvolume -n {{ operator_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ cv_name.stdout }}")].spec.consistencyFactor}{"\n"}'
+         args:
+           executable: /bin/bash
+         register: after_cf
+         failed_when: "after_cf.stdout != cf.stdout"
+
+       - name: Obtain the desiredReplicationFactor after cvr migration and compare with desiredReplication factor obtained before migrate cvr
+         shell: >
+           kubectl get cstorvolume -n {{ operator_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ cv_name.stdout }}")].spec.desiredReplicationFactor}{"\n"}'
+         args:
+           executable: /bin/bash
+         register: after_drf
+         failed_when: "after_drf.stdout != drf.stdout"
+     
+       - name: Get istgt target pod details
+         shell: >
+           kubectl get pods  -l openebs.io/persistent-volume={{ pv.stdout }} 
+           -n {{ operator_ns }} -o custom-columns=:metadata.name --no-headers
+         register: istgt_replica
+
+       - name: Create an empty list of replica pods.
+         set_fact:
+           cstor_replicas_pod : []         
+
+       - name: Obtaining the pool deployments from cvr
+         shell: >
+            kubectl get cvr -n {{ operator_ns }}
+            -l openebs.io/persistent-volume={{ pv.stdout }} --no-headers
+            -o=jsonpath='{range .items[*]}{.metadata.labels.cstorpool\.openebs\.io\/name}{"\n"}{end}'
+         args:
+            executable: /bin/bash
+         register: pool_deployment
+
+       - name: Obtaining the replicasets corresponding to pool deployements.
+         shell: >
+           kubectl get rs --selector=app=cstor-pool -n {{ operator_ns }} --no-headers
+           -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="{{item}}")].metadata.name}'
+         register: rs_list
+         with_items:
+           - "{{ pool_deployment.stdout_lines }}"
+
+       - name: Obtaining the pool pods
```

Comment:
Instead of multiple steps to get pool pods we can use directly
kubectl get pod -n openebs -l app=cstor-pool,openebs.io/cstor-pool=<csp_name>
CSP Name we are already fetching from
- name: Obtaining the pool deployments from cvr